src/emqtt.erl

```diff
@@ -1784,12 +1815,17 @@ reason_code_name(16#A1) -> subscription_identifiers_not_supported;
 reason_code_name(16#A2) -> wildcard_subscriptions_not_supported;
 reason_code_name(_Code) -> unknown_error.
 
-next_reconnect(#state{connect_timeout = Timeout,
+next_reconnect(#state{connect_timeout = Timeout0,
                       retry_timer = RetryTimer,
-                      keepalive_timer = KeepAliveTimer
+                      keepalive_timer = KeepAliveTimer,
+                      reconnect = Reconnect
                      } = State) ->
     ok = cancel_timer(RetryTimer),
     ok = cancel_timer(KeepAliveTimer),
+    Timeout = case is_integer(Reconnect) andalso Reconnect > 0 of
+                  true -> Reconnect;
+                  false -> Timeout0
+              end,
```

Comment:
Nit. This is definitely better, but I anticipate that an emqtt user would want to control a number of reconnect attempts, and (more importantly) wouldn't expect that aggressive exponential backoff down here. So I'd propose to:
1. Abandon this aggressive exponential backoff.
2. Make `reconnect` option mean`boolnean() | _NumberOfReconnectAttempts :: non_neg_integer()`, and introduce **separate** option `reconnect_timeout`.

Reply:
That's a good point, thanks for your review. I'll correct it now